lib/egauge/power_helpers.rb

```diff
@@ -2,23 +2,20 @@ module Egauge
   module PowerHelpers
     KWH_DIVIDER = 3600000
 
-    def full_day_kwh
-      full_day_kwh = {}
-      response = query("/cgi-bin/egauge-show?#{yesterday_query}")
-      response.headers.each do |header|
-        full_day_kwh[header] = calculate_kwh(response.send(header))
-      end
-      full_day_kwh
+    def full_day_kwh(day = Time.now.end_of_day.to_i)
```

Comment:
`Time.now` doesn't respect timezone - you can use `Time.zone.now` since ActiveSupport is included - also I see mixed usage of `Time.now` and `DateTime.now`

Reply:
update: use `Time.current` - it will work if zone is not defined (Time.zone will crash):
```
def current
  ::Time.zone ? ::Time.zone.now : ::Time.now
end
```
https://apidock.com/rails/Time/current/class